arangod/Agency/Constituent.cpp

```diff
@@ -640,164 +634,162 @@ void Constituent::run() {
     act = _agent->config().active();
   }
 
-  if (size() == 1) {
+  {
     std::lock_guard guard{_termVoteLock};
-    _leaderID = _agent->config().id();
-    LOG_TOPIC("66f72", INFO, Logger::AGENCY)
-        << "Set _leaderID to '" << _leaderID << "' in term " << _term;
-  } else {
-    {
-      std::lock_guard guard{_termVoteLock};
-      LOG_TOPIC("29175", INFO, Logger::AGENCY)
-          << "Setting role to follower in term " << _term;
-      _role = FOLLOWER;
-    }
+    LOG_TOPIC("29175", INFO, Logger::AGENCY)
+        << "Setting role to follower in term " << _term;
+    _role = FOLLOWER;
+  }
 
-    std::chrono::steady_clock::time_point constituentLoopStart;
-
-    while (!this->isStopping()) {
-      constituentLoopStart = std::chrono::steady_clock::now();
-
-      role_t role = _role.load(std::memory_order_relaxed);
-
-      if (role == FOLLOWER) {
-        static double const M = 1.0e6;
-        int64_t a = static_cast<int64_t>(M * _agent->config().minPing() *
-                                         _agent->config().timeoutMult());
-        int64_t b = static_cast<int64_t>(M * _agent->config().maxPing() *
-                                         _agent->config().timeoutMult());
-        int64_t randTimeout = RandomGenerator::interval(a, b);
-        int64_t randWait = randTimeout;
-
-        {
-          std::lock_guard guard{_termVoteLock};
-
-          // in the beginning, pure random, after that, we might have to
-          // wait for less than planned, since the last heartbeat we have
-          // seen is already some time ago, note that this waiting time
-          // can become negative:
-          if (_lastHeartbeatSeen > 0.0) {
-            double now = TRI_microtime();
-            randWait -= static_cast<int64_t>(M * (now - _lastHeartbeatSeen));
-            if (randWait < a) {
-              randWait = a;
-            } else if (randWait > b) {
-              randWait = b;
-            }
+  std::chrono::steady_clock::time_point constituentLoopStart;
+
+  while (!this->isStopping()) {
+    constituentLoopStart = std::chrono::steady_clock::now();
+
+    role_t role = _role.load(std::memory_order_relaxed);
+
+    if (role == FOLLOWER) {
+      static double const M = 1.0e6;
+      int64_t a = static_cast<int64_t>(M * _agent->config().minPing() *
+                                       _agent->config().timeoutMult());
+      int64_t b = static_cast<int64_t>(M * _agent->config().maxPing() *
+                                       _agent->config().timeoutMult());
+      int64_t randTimeout = RandomGenerator::interval(a, b);
+      int64_t randWait = randTimeout;
+
+      // Speedup for size() == 1 - do not wait at all. Candidate immediately.
+      if (size() == 1) {
+        LOG_TOPIC("31ed7", INFO, Logger::AGENCY)
+            << "Immediately candidate because of single agency";
```

Comment:
nit: I guess "to candidate" is a rather literal translation. Should rather be "stand(ing) as candidate"

Reply:
```suggestion
      // Speedup for size() == 1 - do not wait at all. Start election immediately.
      if (size() == 1) {
        LOG_TOPIC("31ed7", INFO, Logger::AGENCY)
            << "Immediately start election because of single agency";
```